collie/interactions/datasets.py

```diff
@@ -342,62 +347,63 @@ def __init__(self,
+        # ``positive_set`` attribute is created, a ``set`` of the ``mat`` object that enables
         # fast, O(1), ``(row, col)`` lookup. When ``__getitem__`` is called, negative item or
         # user IDs are sampled one-at-a-time from all possible values in ``self.num_items``
         # or ``self.num_users``, we check if that user ID, item ID pair is in
-        # ``self.positive_sets``, and sample continuously until we have a negative match or
+        # ``self.positive_set``, and sample continuously until we have a negative match or
         # reach a limit of ``max_number_of_samples_to_consider`` tries
         if self.check_num_negative_samples_is_valid:
             print('Checking ``num_negative_samples`` is valid...')
             if self.negative_sample_type == 'item':
                 if self._checked_num_negative_item_samples_is_valid is False:
-                    self._checked_num_negative_item_samples_is_valid = (
-                        self._check_num_negative_samples_is_valid()
-                    )
+                    self._check_num_negative_samples_is_valid(self.negative_sample_type)
+                    self._checked_num_negative_item_samples_is_valid = True
             elif self.negative_sample_type == 'user':
                 if self._checked_num_negative_user_samples_is_valid is False:
-                    self._checked_num_negative_user_samples_is_valid = (
-                        self._check_num_negative_samples_is_valid()
-                    )
+                    self._check_num_negative_samples_is_valid(self.negative_sample_type)
+                    self._checked_num_negative_user_samples_is_valid = True
 
-        self.positive_sets = {}
+        self.positive_set = {}
         if self.max_number_of_samples_to_consider > 0:
             print(f'Generating positive {self.negative_sample_type}s set...')
             self._generate_positive_set()
 
-    def _check_num_negative_samples_is_valid(self) -> bool:
+    def _check_num_negative_samples_is_valid(
+        self,
+        negative_sample_type: Literal['user', 'item']
+    ) -> bool:
         """Check if the number of negative samples is valid."""
-        if self.negative_sample_type == 'item':
+        if negative_sample_type == 'item':
             counter = collections.Counter(self.mat.row)
             num_ids = self.num_items
             opposite_type = 'user'
-        elif self.negative_sample_type == 'user':
+        elif negative_sample_type == 'user':
             counter = collections.Counter(self.mat.col)
             num_ids = self.num_users
             opposite_type = 'item'
 
         max_number_interacted_with = counter.most_common(1)[0][1]
         print(
-            f'Maximum number of {self.negative_sample_type}s a {opposite_type} has interacted with:'
+            f'Maximum number of {negative_sample_type}s a {opposite_type} has interacted with:'
```

Comment:
[Nitpicky] If `opposite_type = 'vowel'`, the `a` in this sentence will need to be an `an`. 
```suggestion
            f'Maximum number of {negative_sample_type}s {'a' if opposite_type == 'user' else 'an'} {opposite_type} has interacted with:'
```